Lab3/ComputerStore.WebUI/ApiControllers/CategoriesApiController.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using ComputerStore.BusinessLogicLayer.Managers;
+using ComputerStore.BusinessLogicLayer.Models;
+using ComputerStore.WebUI.AppConfiguration;
+using ComputerStore.WebUI.Controllers;
+using ComputerStore.WebUI.Mappers;
+using ComputerStore.WebUI.Models;
+using ComputerStore.WebUI.Models.JwtToken;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+namespace ComputerStore.WebUI.ApiControllers
+{
+    [Route("api/categories")]
+    [ApiController]
+    [Authorize(Roles = RolesNames.Admin, AuthenticationSchemes = JwtInfo.AuthSchemes)]
+    public class CategoriesApiController : ControllerBase
+    {
+        private readonly CategoryManager _categoryManager;
+        private readonly ILogger<CategoriesApiController> _logger;
+
+        public CategoriesApiController(CategoryManager categoryManager, 
+                                       ILogger<CategoriesApiController> logger)
+        {
+            _categoryManager = categoryManager;
+            _logger = logger;
+        }
+
+        [HttpGet("categories")]
```

Comment:
Please, avoid overriding routes via HttpGetAttribute and etc.
It would be nice to have the next endpoints (like REST principles):
GET: api/v1/Categories (get all)
GET: api/v1/Categories/{id} (needs RouteAttribute)
POST: api/v1/Categories(with body)
DELETE: api/v1/Categories/{id} (needs RouteAttribute)
PUT: api/v1/Categories (with body)

Fix all controllers.